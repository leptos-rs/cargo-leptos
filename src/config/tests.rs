use super::Config;

fn opts(project: Option<&str>) -> crate::config::Opts {
    crate::config::Opts {
        release: false,
        js_minify: false,
        precompress: false,
        hot_reload: false,
        project: project.map(|s| s.to_string()),
        verbose: 0,
        features: Vec::new(),
        bin_features: Vec::new(),
        lib_features: Vec::new(),
        bin_cargo_args: None,
        lib_cargo_args: None,
        wasm_debug: false,
    }
}

// this test causes issues in CI because the tailwind tmp_file field is an absolute path,
// so differs by platform
/* #[test]
fn test_project() {
    let cli = opts(None);

    let conf = Config::test_load(cli, "examples", "examples/project/Cargo.toml", true);

    insta::assert_debug_snapshot!(conf);
} */

#[test]
fn test_workspace() {
    let cli = opts(None);

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = opts(Some("project1"));

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_in_subdir_project2() {
    let cli = opts(None);

    let conf = Config::test_load(
        cli,
        "examples/workspace/project2",
        "examples/workspace/Cargo.toml",
        true,
        None,
    );

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_bin_args_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::test_load(
        cli,
        "examples",
        "examples/workspace/Cargo.toml",
        true,
        Some(&["--".to_string(), "--foo".to_string()]),
    );

    insta::assert_debug_snapshot!(conf);
}
