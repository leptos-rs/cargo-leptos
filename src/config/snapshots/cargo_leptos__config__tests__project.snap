---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "example",
            lib: LibPackage {
                name: "example",
                rel_dir: ".",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/example.wasm",
                    dest: "target/site/pkg/example.wasm",
                    site: "pkg/example.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/pkg/example.js",
                    site: "pkg/example.js",
                },
                features: [
                    "hydrate",
                ],
                default_features: false,
                output_name: "example",
                src_paths: "src",
                profile: Debug,
                tailwind: false,
                ..
            },
            bin: BinPackage {
                name: "example",
                rel_dir: ".",
                exe_file: "target/server/debug/example",
                target: "example",
                features: [
                    "ssr",
                ],
                default_features: false,
                src_paths: "src",
                profile: Debug,
                ..
            },
            style: Some(
                StyleConfig {
                    file: SourcedSiteFile {
                        source: "style/main.scss",
                        dest: "target/site/pkg/example.css",
                        site: "pkg/example.css",
                    },
                    browserquery: "defaults",
                },
            ),
            watch: true,
            release: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: Some(
                 {
                    cmd: "npx playwright test",
                    dir: "end2end",
                },
            ),
            assets: Some(
                AssetsConfig {
                    dir: "assets",
                },
            ),
            ..
        },
    ],
    cli: Opts {
        release: false,
        project: None,
        features: [],
        lib_features: [],
        bin_features: [],
        verbose: 0,
    },
    watch: true,
    ..
}
